tools/bwa.py

```diff
@@ -46,18 +48,28 @@ def index(self, inFasta, algorithm=None):
         cmd.append(inFasta)
         self.execute('index', cmd)
 
-    def mem(self, inReads, refDb, outAlign, threads=None):
+    def mem(self, inReads, refDb, outAlign, opts=None, threads=None):
+        opts = [] if not opts else opts
+
         threads = threads or util.misc.available_cpu_count()
         samtools = tools.samtools.SamtoolsTool()
         fq1 = util.file.mkstempfname('.1.fastq')
         fq2 = util.file.mkstempfname('.2.fastq')
         aln_sam = util.file.mkstempfname('.sam')
+        aln_sam_sorted = util.file.mkstempfname('sorted.sam')
         samtools.bam2fq(inReads, fq1, fq2)
-        self.execute('mem', ['-t', str(threads), refDb, fq1, fq2], stdout=aln_sam)
+        self.execute('mem', opts + ['-t', str(threads), refDb, fq1, fq2], stdout=aln_sam)
         os.unlink(fq1)
         os.unlink(fq2)
-        samtools.sort(aln_sam, outAlign)
+        samtools.sort(aln_sam, aln_sam_sorted)
         os.unlink(aln_sam)
-        samtools.index(outAlign)
+        # cannot index sam files; only do so if a bam is desired
+        if outAlign.endswith(".bam") or outAlign.endswith(".cram"):
+            # convert sam -> bam
+            samtools.view(["-b"], aln_sam_sorted, outAlign)
```

Comment:
Can we also get rid of this sam->bam conversion here (with the aln_sam_sorted temp file created above)? Takes a lot of excess tempdir disk space and runtime compared to before (where samtools.sort did the sam->bam conversion during sorting).  My suggestion would be to remove the aln_sam_sorted temp file, revert the samtools.sort to write directly to outAlign, and put only the samtools.index command in the if block (with nothing else, and no elif/else block)